torch/_dynamo/variables/user_defined.py

```diff
@@ -369,6 +370,16 @@ def call_function(
                 args[0],
                 mutable_local=MutableLocal(),
             )
+        elif is_typeddict(self.value):
+            if self.value.__optional_keys__:
+                unimplemented("TypedDict with optional keys not supported")
+            if len(args):
+                unimplemented("TypedDict only supports kwargs")
+            dict_items = {
+                variables.ConstantVariable.create(key): value
+                for key, value in kwargs.items()
+            }
+            return variables.ConstDictVariable(dict_items, mutable_local=MutableLocal())
```

Comment:
The `__new__` method for a TypedDict class is `dict.__new__`. It supports `TypedDictClass(mapping, **kwargs)` and `TypedDictClass(seq2, **kwargs)`.

```python
In [1]: from typing import TypedDict

In [2]: class MyDict(TypedDict):
   ...:     foo: int
   ...:     bar: float
   ...:     

In [3]: MyDict([('foo', 1), ('bar', 0.0)])
Out[3]: {'foo': 1, 'bar': 0.0}

In [4]: MyDict({'foo': 1, 'bar': 0.0})
Out[4]: {'foo': 1, 'bar': 0.0}

In [5]: MyDict([('foo', 1)], bar=10.0)
Out[5]: {'foo': 1, 'bar': 10.0}
```

Just use `dict.__new__` here:

```suggestion
            return variables.BuiltinVariable(dict).call_dict(tx, args, kwargs)
```